core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanContext.scala

```diff
@@ -53,6 +56,9 @@ trait SpanContext {
     * parent.
     */
   def isRemote: Boolean
+
+  def storeInContext(context: Vault): Vault =
```

Comment:
This state was previously part of `Scope`, managed by `TraceScope`.  Now we should keep this key and `TraceScope` synchronized, but this one is public.  We need that internal Java state to make it work, but we need the public state for propagation. 🤔